### DISCUSSION TITLE
Support for Google Cast

### DISCUSSION THREAD

**Discussion by raress96** [music-assistant/support#491] [Feature requests and ideas]
Hello,

I would like to be able to cast from Google to Music Assistant like it was a speaker. To use it like a much more powerful alternative to the Google Chromecast Audio.

So I could say "Hey Google, play music on Home Assistant" or something similar, and then it will cast to Music Assistant which will then forward the audio to whatever speakers I want.

This will provide a lot of flexibility, Youtube Music will work, other apps that use casting will also work and you could control it using voice commands with Google Assistant.

I think you will lose the queue system and playlist from Music Assistant, so only the playback controls and speaker controls would be available in this mode.

Would you think it would be possible for Music Assistant to be seen as a speaker by Google so media could be casted to it?

Interested if others would also like something like this or if there is another solution for this.

**Comment by marcelveldt** [comment]
That is not possible I'm afraid because that part of Google cast is proprietary.
Having MA as Spotify connect target is possible (and another feature request somewhere in the list). Making MA a cast target itself is technically not/never possible.

I do however sense another wish in your request and that is the ability to send voice commands to Music Assistant.
That is in fact possible. In fact you can already do that now with Home Assistant connected to Google Assistant and a little creativity with scripts named as playlists etc.

A super upgrade would be to talk to Music Assistant directly from Google Assistant such as "Hey Google, ask Music Assistant to play My Playlist on the Office speaker" or something like that.

**Reply by raress96** [reply]
I see, was afraid that google cast would not work...

I saw that in the other discussion regarding Spotify it's mentioned for MA to act as a relay, so I think it would still be helpful to have inputs to MA (which could be bluetooth, 3.5mm jack or anything) and then using MA as a proxy that forwards the input to respective speakers. Using HA automations with this will make it quite powerful. (and I could even hook up a google chromecast audio to the input and have google cast somewhat work)

EDIT: While digging around a bit, I came across this:
https://developers.google.com/cast/docs/web_receiver#default_media_web_receiver
Which to me seems that building a Cast Receiver is possible, at least at first glance. Will have to look further into it.

**Comment by OzGav** [comment]
This is an outdated discussion that relates to v1
